### ISSUE TITLE
Add a @nx/web:library generator

### ISSUE THREAD

**Issue by chriswoodie** [nrwl/nx#17815]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description
Add a @nx/web:library generator to support creating a web components library rather than an application.

## Motivation
Currently there is a @nx/web:application that creates the application in the `apps` folder, this is fine if you're making an application but in my case I'm using it as a library that other applications will depend on. However, there is no way to create a library using @nx/web currently, I'm not sure if the `directory` parameter is supposed to be used here, but it seems wrong.

## Suggested Implementation
Add a @nx/web:library generator which creates a new shareable library in the `library` folder. This should also automatically add a new item to `paths` in the `tsconfig.base.json` file, such as:

```
"paths": {
      "@<project-name>/<library-name>": ["libs/<library-name>/src/index.ts"],
}
```

Basically this should mirror the behaviour of creating an Angular library or other, which currently works as expected.

## Alternate Implementations
None that I can think of.

**If I'm going about this in the wrong way, then please inform me how to create a Web Components library using a different Nx command.**

**Comment by Coly010** [comment]
You could achieve this by using the `@nx/js:library` generator.

The JS library generator creates a basic TS/JS where you can do anything supported by JS or TS.

**Comment by skog-newglue** [comment]
@Coly010 Yep this is what I ended up doing just now. I think it should perhaps be noted in the docs to make it clearer.
